SDL audio implementation (mk 2)

Phlamethrower opened the pull request:
This is https://github.com/ArcEm-emu/ArcEm/pull/12 with additional changes to fix a couple of issues with the core sound code and significantly reduce the number of audio breakups.

A couple of potential future improvements:

* Try and minimise the latency (tricky because HostFS can block for arbitrary amounts of time)
* Fix the core code so that when sound DMA is disabled it sends silence instead of nothing

ccawley2011 commented:
Thanks, I've had the opportunity to test this out - it works nicely on newer machines, but on older ones the sound continues to break up quite a bit. It's still an improvement over PR #12 though, so it should be OK to merge once it's ready. Sorry for taking a while to get back to you on this.

ccawley2011 commented on SDL/sound.c:
```diff
@@ -0,0 +1,319 @@
+  {
+    fprintf(logfile,"event,time,used,delta\n");
+  }
+  fprintf(stderr, "Timer frequency %llu\n", SDL_GetPerformanceFrequency());
+  logt0 = SDL_GetPerformanceCounter();
```
I know this is a small thing since it's only used for debugging, but could you add a basic fallback for these two functions when building with SDL 1.2?

Phlamethrower replied:
Good catch - fixed now.

ccawley2011 commented:
When you next get the opportunity, could you rebase this and update the CMake project?

In addition, would you rather I merged PR #19 now so that you can adapt this PR for it, or for me to hold off on that until this is ready, and then I adapt that PR to work with these changes?

Phlamethrower commented:
* Rebased onto latest main and fixed up
* CMake support added
* Warnings squashed

> In addition, would you rather I merged PR https://github.com/ArcEm-emu/ArcEm/pull/19 now so that you can adapt this PR for it, or for me to hold off on that until this is ready, and then I adapt that PR to work with these changes?

Are there any changes you'd like to see to this sound code before it gets merged? If not, then we should probably merge it now. Better than having it sit in a PR forever 😅 

I'd have to do some setup to get an SDL3 build environment (my Linux machine is long overdue a refresh/upgrade), so if this code gets merged it would probably be quicker for you to handle the SDL3 update

ccawley2011 commented:
> I'll give this PR another quick test run, and if there are no other issues I'll merge it.

I've tested this and it looks/sounds OK, so I've merged it. Thanks for your work on this.

Phlamethrower commented:
No problem!
